Microsoft.Toolkit.Uwp/Helpers/SystemInformation.cs

```diff
@@ -337,6 +349,7 @@ private bool DetectIfAppUpdated()
             else
             {
                 var lastVersion = _localObjectStorageHelper.Read<string>(nameof(currentVersion));
```

Comment:
I'm a bit confused by this point. `PreviousVersionInstalled` is only installed in a conditional branch, which from what I can see will not be used the first time an installed app is launched, as the target setting key `"currentVersion"` will not be found. Isn't this a bug, or am I missing something here? 🤔

Reply:
On Line 333 if the key doesn't exist we save the `ApplicationVersion` of the app, so on first install `PreviousVersionInstalled` will match the application and `FirstVersionInstalled`.

`PreviousVersionInstalled` will only be meaningful the first time after you update, just like the `IsAppUpdated` property.

Reply:
> "so on first install PreviousVersionInstalled will match the application and FirstVersionInstalled."

How though? From what I can see, on first launch we do indeed save the current version, but then never set `FirstVersionInstalled`, as that assignment is in the other branch that's not executed in this case. So wouldn't `FirstVersionInstalled` just remain the default value of `0.0.0.0` on first launch? 🤔

> "PreviousVersionInstalled will only be meaningful the first time after you update"

I can agree with that, I was just confused though on whether we still wanted that initial value for the first launch to match the current application version instead of just being 0 (unless I'm reading the code wrong here?).

Reply:
Ah, sorry I misread the branch. `FirstVersionInstalled` is independent anyway (and set automatically in `DetectFirstVersionInstalled`), but will be correct on first run; `PreviousVersionInstalled` will be `0.0.0.0` I think the first run as the conditional branch it's set in is skipped, right?

Reply:
As per https://github.com/windows-toolkit/WindowsCommunityToolkit/pull/4000/commits/76cf98838469d1f531022fbc23637cbf073bbcc4, the behavior of this property is always well defined and it will just return the same version as the current one if checked at the first startup and/or when there hasn't been an update since last time the app was opened 🙂